Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -623,17 +618,19 @@ private void OnSolutionClosing() {
         }
 
         private void OnSolutionOpened() {
-            if (_workspaceFoldersSupported && IsInitialized && !_sentInitialWorkspaceFolders) {
-                _sentInitialWorkspaceFolders = true;
-                JoinableTaskContext.Factory.RunAsync(async () => {
-                    // If workspace folders are supported, then send our workspace folders
-                    var folders = from n in this.ProjectContextProvider.ProjectNodes
-                                  select new WorkspaceFolder { uri = new System.Uri(n.BaseURI.Directory), name = n.Name };
-                    this._workspaceFolders = new List<WorkspaceFolder>(folders);
-                    if (folders.Any()) {
-                        await InvokeDidChangeWorkspaceFoldersAsync(folders.ToArray(), new WorkspaceFolder[0]);
-                    }
-                });
+            if (_workspaceFoldersSupported && IsInitialized && !_sentInitialWorkspaceFolders) { 
+                JoinableTaskContext.Factory.RunAsync(OnSolutionOpenedAsync);
+            }
+        }
+
+        private async Task OnSolutionOpenedAsync() {
```

Comment:
made this async so that we can wait for the folder update "InvokeDidChangeWorkspaceFoldersAsync"